core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -628,8 +660,38 @@ class DAGScheduler(
       callSite: CallSite,
       resultHandler: (Int, U) => Unit,
       properties: Properties): Unit = {
+    runJob[T, U](rdd, (_: Int) => {}, func, partitions, callSite, resultHandler, properties)
+  }
+
+  /**
+   * Create configs on driver as part of commit protocol using a provided function that accepts
+   * an integer (stageId) as parameter and then run an action job on the given RDD and pass all
+   * the results to the resultHandler function as they arrive.
+   *
+   * @param rdd target RDD to run tasks on
+   * @param jobSetupFunc This function gets called on the driver after the final stage is known.
+   * Its primary purpose is as part of the commit protocol, since that needs
+   * to know the stageId for creating a unique id. See SparkHadoopWriter
+   * @param func a function to run on each partition of the RDD
+   * @param partitions set of partitions to run on; some jobs may not want to compute on all
+   *   partitions of the target RDD, e.g. for operations like first()
+   * @param callSite where in the user program this job was called
+   * @param resultHandler callback to pass each result to
+   * @param properties scheduler properties to attach to this job, e.g. fair scheduler pool name
+   *
+   * @note Throws `Exception` when the job fails
+   */
+  private[spark]
+  def runJob[T, U](
```

Comment:
same thing here, don't need a new method

Reply:
I added this, but chnaged the original one to call this at line 663